### PR TITLE
Remove unnecessary local variables for return values

### PR DESCRIPTION
**Pull request by peterjc**
This pull request addresses a minor style issue, using another flake8 plugin to enforce this in future (code PIE781 from flake8-pie for assign-and-return).

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Scripts/xbbtools/xbb_blast.py [hunk 1]
```diff
@@ -116,9 +116,7 @@ def get_blast_binaries(self):
 
     def database_readable(self, db_paths):
         """Return the name of the blast database without path and extension."""
```
Comment (MarkusPiotrowski) [inline Scripts/xbbtools/xbb_blast.py]: Unrelated to this PR, but this should better be "Return list of blast database names without path and extension."

**Comment by peterjc** [comment]
Its a good point Markus, which I did have in mind when making these edits. Where the temporary local variable had a meaningful name, I felt there was no harm removing it given the function name and/or docstring. I could have missed something though.

Quite a few of these files are on modules @mdehoon knows best, so will wait for his comments.

**Comment by peterjc** [comment]
In this case with the two functions returning ``record`` (singular) and ``records`` (plural), I can see some advantage in retaining the local variable names. Reverted.

**Comment by peterjc** [comment]
If with the ``>>>`` bit you were alluding to doctest examples, given their primary role as documentation, including temporary return variables with meaningful examples will often make sense.

However, currently we don't run python code style checks on the doctest code itself (not sure if there is an easy way to do this or not, it could be useful).

**Comment by mdehoon** [comment]
@peterjc  No, I wasn't thinking of the doctests in particular; I was just thinking for developers the code is easier to understand with the explicit variable names.

**Comment by peterjc** [comment]
@mdehoon I don't feel strongly enough on this stylistic point to push for it to be automatically enforced if some of our key contributors (like you) object.

I could cut this back to just removing the local return variables outside of ``Bio/Entrez`` (and remove the automatic enforcement changes)?

**Comment by peterjc** [comment]
Updated to exclude the changes in ``Bio/Entrez``, and not to enforce this automatically.

**Comment by peterjc** [comment]
@mdehoon is this OK now, or should I close this?

**Comment by peterjc** [comment]
This has built up lots of conflicts now - closing.

**Pull request closed without merging**